Separate keystore dir per multiaccount

rasom opened the pull request:
As part of https://github.com/status-im/status-react/issues/10195, having all keys in a separate dir will simplify account removing. Also it will allow to have same accounts to added via private key/seed in wallet in different multiaccs (as keys will be stored in separate dirs).

This commit provides `MigrateKeyStoreDir` method which moves multiacc's keys from a common keystore directory to multiacc specific directory.

status: ready

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (60)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 17ec9fec | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/1/) | 2020-06-18 10:30:39 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200618-102818-17ec9f.zip) |
| :heavy_check_mark: | 17ec9fec | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/1/) | 2020-06-18 10:33:49 | ~5 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200618-102816-17ec9f.zip) |
| :heavy_check_mark: | 17ec9fec | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/1/) | 2020-06-18 10:35:59 | ~7 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200618-102815-17ec9f.aar) |
| | | | | | | |
| :heavy_check_mark: | 5fe71990 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/2/) | 2020-06-22 12:07:42 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200622-120508-5fe719.zip) |
| :heavy_check_mark: | 5fe71990 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/2/) | 2020-06-22 12:09:14 | ~4 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200622-120507-5fe719.zip) |
| :heavy_check_mark: | 5fe71990 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/2/) | 2020-06-22 12:13:02 | ~7 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200622-120505-5fe719.aar) |
| | | | | | | |
| :heavy_check_mark: | 9299693f | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/3/) | 2020-06-22 15:37:42 | ~7 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200622-153018-929969.zip) |
| :heavy_check_mark: | 9299693f | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/3/) | 2020-06-22 15:40:04 | ~9 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200622-153016-929969.zip) |
| :heavy_check_mark: | 9299693f | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/3/) | 2020-06-22 15:42:05 | ~11 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200622-153014-929969.aar) |
| | | | | | | |
| :heavy_check_mark: | ca7e5760 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/4/) | 2020-06-23 15:57:47 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200623-155530-ca7e57.zip) |
| :heavy_check_mark: | ca7e5760 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/4/) | 2020-06-23 15:58:34 | ~3 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200623-155528-ca7e57.zip) |
| :heavy_check_mark: | ca7e5760 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/4/) | 2020-06-23 16:02:45 | ~7 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200623-155526-ca7e57.aar) |
| | | | | | | |
| :x: | cb25fcf1 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/5/) | 2020-06-23 16:02:39 | ~26 sec | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/5/consoleText) |
| :x: | cb25fcf1 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/5/) | 2020-06-23 16:03:08 | ~53 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/5/consoleText) |
| :x: | cb25fcf1 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/5/) | 2020-06-23 16:04:20 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/5/consoleText) |
| | | | | | | |
| :heavy_check_mark: | a25159d1 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/6/) | 2020-06-23 16:44:14 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200623-164138-a25159.zip) |
| :heavy_check_mark: | a25159d1 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/6/) | 2020-06-23 16:44:22 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200623-164136-a25159.zip) |
| :heavy_check_mark: | a25159d1 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/6/) | 2020-06-23 16:48:28 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200623-164134-a25159.aar) |
| | | | | | | |
| :heavy_check_mark: | e7184920 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/7/) | 2020-06-24 15:01:36 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200624-145925-e71849.zip) |
| :heavy_check_mark: | e7184920 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/7/) | 2020-06-24 15:02:12 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200624-145923-e71849.zip) |
| :heavy_check_mark: | e7184920 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/7/) | 2020-06-24 15:06:23 | ~7 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200624-145921-e71849.aar) |
| | | | | | | |
| :x: | 53678439 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/8/) | 2020-07-01 14:13:56 | ~1 min | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/8/consoleText) |
| :x: | 53678439 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/8/) | 2020-07-01 14:14:14 | ~1 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/8/consoleText) |
| :x: | 53678439 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/8/) | 2020-07-01 14:14:23 | ~2 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/8/consoleText) |
| | | | | | | |
| :heavy_check_mark: | 619f17cf | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/9/) | 2020-07-01 14:26:15 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200701-142426-619f17.zip) |
| :heavy_check_mark: | 619f17cf | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/9/) | 2020-07-01 14:28:30 | ~3 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200701-142424-619f17.zip) |
| :heavy_check_mark: | 619f17cf | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/9/) | 2020-07-01 14:30:31 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200701-142423-619f17.aar) |
| :heavy_check_mark: | 619f17cf | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/12/) | 2020-07-02 03:13:51 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200702-031156-619f17.zip) |
| :heavy_check_mark: | 619f17cf | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/12/) | 2020-07-02 03:14:35 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200702-031154-619f17.zip) |
| :heavy_check_mark: | 619f17cf | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/12/) | 2020-07-02 03:17:45 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200702-031151-619f17.aar) |
| | | | | | | |
| :heavy_check_mark: | 6034ae61 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/10/) | 2020-07-02 01:59:09 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200702-015716-6034ae.zip) |
| :heavy_check_mark: | 6034ae61 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/10/) | 2020-07-02 02:02:08 | ~4 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200702-015715-6034ae.zip) |
| :heavy_check_mark: | 6034ae61 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/10/) | 2020-07-02 02:03:06 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200702-015713-6034ae.aar) |
| | | | | | | |
| :heavy_check_mark: | a59b68f3 | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/11/) | 2020-07-02 02:47:11 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200702-024524-a59b68.zip) |
| :heavy_check_mark: | a59b68f3 | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/11/) | 2020-07-02 02:48:38 | ~3 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200702-024522-a59b68.zip) |
| :heavy_check_mark: | a59b68f3 | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/11/) | 2020-07-02 02:51:03 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200702-024520-a59b68.aar) |
| | | | | | | |
| :heavy_check_mark: | 4a6bb60c | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/13/) | 2020-07-02 03:15:32 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200702-031353-4a6bb6.zip) |
| :heavy_check_mark: | 4a6bb60c | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/13/) | 2020-07-02 03:18:19 | ~3 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200702-031438-4a6bb6.zip) |
| :heavy_check_mark: | 4a6bb60c | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/13/) | 2020-07-02 03:23:23 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200702-031748-4a6bb6.aar) |
| | | | | | | |
| :heavy_check_mark: | e6cf3122 | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/14/) | 2020-07-06 13:08:23 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200706-130552-e6cf31.zip) |
| :heavy_check_mark: | e6cf3122 | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/14/) | 2020-07-06 13:09:17 | ~3 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200706-130554-e6cf31.zip) |
| :heavy_check_mark: | e6cf3122 | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/14/) | 2020-07-06 13:12:13 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200706-130550-e6cf31.aar) |
| | | | | | | |
| :x: | 794214c6 | [#15](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/15/) | 2020-07-08 08:22:37 | ~29 sec | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/15/consoleText) |
| :x: | 794214c6 | [#15](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/15/) | 2020-07-08 08:23:34 | ~1 min | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/15/consoleText) |
| :x: | 794214c6 | [#15](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/15/) | 2020-07-08 08:24:09 | ~2 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/15/consoleText) |
| | | | | | | |
| :heavy_check_mark: | c070b153 | [#16](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/16/) | 2020-07-08 08:30:01 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200708-082823-c070b1.zip) |
| :heavy_check_mark: | c070b153 | [#16](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/16/) | 2020-07-08 08:30:52 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200708-082821-c070b1.zip) |
| :heavy_check_mark: | c070b153 | [#16](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/16/) | 2020-07-08 08:34:26 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200708-082819-c070b1.aar) |
| | | | | | | |
| :heavy_check_mark: | 55499871 | [#17](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/17/) | 2020-07-08 08:41:00 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200708-083928-554998.zip) |
| :heavy_check_mark: | 55499871 | [#17](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/17/) | 2020-07-08 08:41:55 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200708-083926-554998.zip) |
| :heavy_check_mark: | 55499871 | [#17](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/17/) | 2020-07-08 08:44:59 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200708-083924-554998.aar) |
| | | | | | | |
| :heavy_check_mark: | 96878655 | [#18](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/18/) | 2020-07-08 09:09:07 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200708-090710-968786.zip) |
| :heavy_check_mark: | 96878655 | [#18](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/18/) | 2020-07-08 09:12:47 | ~5 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200708-090708-968786.zip) |
| :heavy_check_mark: | 96878655 | [#18](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/18/) | 2020-07-08 09:14:44 | ~7 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200708-090707-968786.aar) |
| | | | | | | |
| :heavy_check_mark: | 0320e062 | [#19](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/19/) | 2020-07-08 09:57:24 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200708-095537-0320e0.zip) |
| :heavy_check_mark: | 0320e062 | [#19](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/19/) | 2020-07-08 09:58:15 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200708-095535-0320e0.zip) |
| :heavy_check_mark: | 0320e062 | [#19](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/19/) | 2020-07-08 10:01:14 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200708-095532-0320e0.aar) |
| | | | | | | |
| :heavy_check_mark: | 8803abe0 | [#20](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/20/) | 2020-07-08 10:01:53 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200708-100008-8803ab.zip) |
| :heavy_check_mark: | 8803abe0 | [#20](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/20/) | 2020-07-08 10:02:28 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200708-100006-8803ab.zip) |
| :heavy_check_mark: | 8803abe0 | [#20](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/20/) | 2020-07-08 10:06:45 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200708-100116-8803ab.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 65dfc34b | [#21](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/21/) | 2020-07-08 10:12:05 | ~1 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200708-101021-65dfc3.zip) |
| :heavy_check_mark: | 65dfc34b | [#21](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/21/) | 2020-07-08 10:14:07 | ~3 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200708-101020-65dfc3.zip) |
| :heavy_check_mark: | 65dfc34b | [#21](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/21/) | 2020-07-08 10:16:30 | ~6 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200708-101018-65dfc3.aar) |
| | | | | | | |
| :heavy_check_mark: | 6d5a9328 | [#22](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1993/22/) | 2020-07-09 09:44:20 | ~2 min | `linux` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-200709-094217-6d5a93.zip) |
| :heavy_check_mark: | 6d5a9328 | [#22](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1993/22/) | 2020-07-09 09:44:48 | ~2 min | `ios` | [:package:`zip`](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-200709-094216-6d5a93.zip) |
| :heavy_check_mark: | 6d5a9328 | [#22](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1993/22/) | 2020-07-09 09:47:54 | ~5 min | `android` | [:package:`aar`](https://status-go.ams3.digitaloceanspaces.com/status-go-android-200709-094214-6d5a93.aar) |

rasom committed:
Separate keystore directory per multiacc

This commit provides `MigrateKeyStoreDir` method which moves multiacc's
keys from a common keystore directory to multiacc specific directory.

andremedeiros commented on api/geth_backend.go:
```diff
@@ -305,6 +308,90 @@ func (b *GethStatusBackend) startNodeWithAccount(acc multiaccounts.Account, pass
+	return paths, nil
+}
+
+func (b *GethStatusBackend) MigrateKeyStoreDir(acc multiaccounts.Account, password, oldDir, newDir string) error {
```
Is this supposed to be idempotent?

rasom replied:
Yep it safe to be called many times (if that's what you are asking about).

cammellos commented on api/geth_backend.go:
```diff
@@ -305,6 +308,90 @@ func (b *GethStatusBackend) startNodeWithAccount(acc multiaccounts.Account, pass
+			return nil
+		}
+
+		rawKeyFile, e := ioutil.ReadFile(path)
```
`err` is more idiomatic, instead of `e`

rasom replied:
fixed

cammellos commented on api/geth_backend.go:
```diff
@@ -305,6 +308,90 @@ func (b *GethStatusBackend) startNodeWithAccount(acc multiaccounts.Account, pass
+			return fmt.Errorf("failed to read key file: %s", e)
+		}
+
+		address := types.HexToAddress("0x" + accountKey.Address).Hex()
```
Not sure I understand here, are you basically normalizing the address by parsing it and converting it back to hex? 
If so I would leave a comment

Samyoul replied:
Is this a kind of validation of the address?

rasom replied:
I actually just copied code from here https://github.com/status-im/status-go/blob/8b61d92d5834215fb06532c527b0bead7b4625d9/account/accounts.go#L142 ` ¯\_(ツ)_/¯`

rasom replied:
@cammellos @Samyoul let me know if there is a better way to write this code, but at least it works

Samyoul replied:
@rasom Would you mind breaking this section of code out into a dedicated `verifyKeyFile()` function and adding some tests specifically for this functionality? 😊 Thank you

rasom replied:
@Samyoul I added one tiny test for `MigrateKeyStoreDir`, it is a bit puzzling for me how to juggle with slice references (I suppose it would be necessary with `verifyKeyFile` )

rasom replied:
let me know if you want to see something else there, meanwhile the whole thing is tested manually here https://github.com/status-im/status-react/pull/10847

cammellos reviewed (approved):
Version bump is missing, left a few minor comments, otherwise looks good

Samyoul commented on api/geth_backend.go:
```diff
@@ -305,6 +308,90 @@ func (b *GethStatusBackend) startNodeWithAccount(acc multiaccounts.Account, pass
+		addresses[account.Address.Hex()] = struct{}{}
+	}
+
+	paths, err := getKeyFilePathsByAddresses(oldDir, addresses)
+
+	if err != nil {
+		return err
+	}
```
Little formatting tweak. It's idiomatic to have the err check directly below the function that generates an error.

```suggestion
	paths, err := getKeyFilePathsByAddresses(oldDir, addresses)
	if err != nil {
		return err
	}
```

rasom replied:
fixed

Samyoul reviewed (changes requested):
Good work. I believe that this code needs some tests adding, as currently there aren't any tests the new functionality added.

Samyoul reviewed (approved):
Thank you very much
